### PR TITLE
bump nano v3.1 & iptraf v3.0.1 with wchar support

### PR DESCRIPTION
**Pull request by stalchuk**
Bump new ipk version with wchar support for:
- nano 3.1
- iptraf 3.0.1

### CODE REVIEW

File: make/ncurses.mk [hunk 1]
```diff
@@ -90,6 +90,7 @@ $(NCURSES_DIR)/.configured: $(DL_DIR)/$(NCURSES_SOURCE) $(NCURSES_PATCHES) make/
 		$(STAGING_LIB_DIR)/libmenu.* \
 		$(STAGING_LIB_DIR)/libncurses.* \
 		$(STAGING_LIB_DIR)/libpanel.* \
+		$(STAGING_LIB_DIR)/libtinfo.* \
```
Comment (alllexx88) [inline make/ncurses.mk]: No point in this addition: no reason to clean `$(STAGING_LIB_DIR)/libtinfo.*` twice.

File: make/ncurses.mk [hunk 2]
```diff
@@ -105,6 +106,7 @@ $(NCURSES_DIR)/.configured: $(DL_DIR)/$(NCURSES_SOURCE) $(NCURSES_PATCHES) make/
 		--host=$(GNU_TARGET_NAME) \
 		--build=$(GNU_HOST_NAME) \
 		--prefix=$(TARGET_PREFIX)	\
+		--with-termlib		\
```
Comment (alllexx88) [inline make/ncurses.mk]: Generating a separate terminfo library will break many Optware packages, so we don't want this.

File: make/nano.mk [hunk 3]
```diff
@@ -4,25 +4,25 @@
...
+NANO_SOURCE=nano-$(NANO_VERSION).tar.xz
 NANO_DIR=nano-$(NANO_VERSION)
-NANO_UNZIP=zcat
+NANO_UNZIP=xzcat
 NANO_MAINTAINER=Mark Donszelmann <mark@donszelmann.com>
 NANO_DESCRIPTION=A pico like editor
 NANO_SECTION=editor
 NANO_PRIORITY=optional
-NANO_DEPENDS=ncurses, zlib, file
+NANO_DEPENDS=ncurses, ncursesw, zlib, file
```
Comment (alllexx88) [inline make/nano.mk]: Do we need a dependency on `ncurses`? `nano` binary is linked against `ncursesw`.

File: make/nano.mk [hunk 4]
```diff
@@ -39,7 +39,7 @@ $(DL_DIR)/$(NANO_SOURCE):
 nano-source: $(DL_DIR)/$(NANO_SOURCE) $(NANO_PATCHES)
 
 $(NANO_BUILD_DIR)/.configured: $(DL_DIR)/$(NANO_SOURCE) $(NANO_PATCHES) make/nano.mk
-	$(MAKE) ncurses-stage zlib-stage file-stage
+	$(MAKE) ncurses-stage ncursesw-stage zlib-stage file-stage
```
Comment (alllexx88) [inline make/nano.mk]: What is `ncurses-stage` needed here for?

**Comment by alllexx88** [comment]
Looks almost good to me. Things to be fixed:

- Remove `nano` dependency on `ncurses`
- Roll back `ncurses` changes: we don't want a separate terminfo library

See changes comments for details.

**Comment by stalchuk** [comment]
Pull request reworked:

- changes on **ncurses** reverted;
- nano & iptraf built with **ncursesw**;

`libtinfo` is required, if **nano** is build with utf-8 support, that is why **ncurses** needs to be build with `--with-termlib` option. Since it can break many packages, I enabled **termlib** for **ncursesw.** Unfortunately, **nano** still looking for `-ltinfo` instead of `-ltinfow` when configured. Traditional way to provide proper library name from the config script didn't work for me, so some dirty fix is applied.

17 other packages depend on `ncursesw,` which need to be rebuild to be linked with `ltinfow.` I'm not sure if it worth it.

Please review.

**Comment by alllexx88** [comment]
This

> libtinfo is required, if nano is build with utf-8 support

is actually wrong. If you look at `nano` sources, you'll see that `libtinfo` isn't hardcoded anywhere. Maybe you're getting `-ltinfo` flag from your host's ncurses development files (like `ncursesw5-config` script). I suggest passing `ac_cv_prog_NCURSESW_CONFIG=$(STAGING_PREFIX)/bin/ncursesw5-config` environment variable to `./configure`. We also don't need `NANO_CPPFLAGS` then: needed flags are provided by the `$(STAGING_PREFIX)/bin/ncursesw5-config` script.

As for `ncurses(w)`, like I said, we don't want a *separate* `libtinfo(w)`. When built without `--with-termlib` flag, `termlib` is embedded in `libncurses(w)`. The only reason to have a separate `termlib` is to be able to link against `libtinfo(w)` directly, without linking with `libncurses(w)`, for dependencies that need `termlib`, but not entire `ncurses(w)`. We don't do this in Optware-ng, so please undo any changes to `ncurses.mk`.

Likewise, please remove `-ltinfow` flag from `iptraf.mk`.

Finally, no need to bump `IPTRAF(NANO)_IPK_VERSION`, like you did in the second commit, since no binaries have yet been uploaded to the feed.

P.S. If you don't mind, I can accept you PR now and fix the nuances I noted myself.

**Comment by stalchuk** [comment]
Cleaning up the code as per your recommendation.
Thank you!

**Comment by alllexx88** [comment]
Thank you! Merging.

**Pull request merged**